src/libcore/task/spawn_error.rs

```diff
@@ -49,3 +50,33 @@ pub struct SpawnObjError {
     /// The task for which spawning was attempted
     pub task: TaskObj,
 }
+
+/// The result of a failed spawn
+#[derive(Debug)]
+pub struct SpawnLocalObjError {
+    /// The kind of error
+    pub kind: SpawnErrorKind,
+
+    /// The task for which spawning was attempted
+    pub task: LocalTaskObj,
+}
+
+impl SpawnLocalObjError {
+    /// Converts the `SpawnLocalObjError` into a `SpawnObjError`
+    /// To make this operation safe one has to ensure that the `UnsafeTask`
+    /// instance from which the `LocalTaskObj` stored inside was created
+    /// actually implements `Send`.
+    pub unsafe fn as_spawn_obj_error(self) -> SpawnObjError {
```

Comment:
What is this for? This seems like a pretty big footgun.

Also, an unsafe function for going from `LocalTaskObj` -> `TaskObj` would give you the same functionality but would be more generally useful IMO.

Reply:
https://github.com/rust-lang-nursery/futures-rs/pull/1046/files#diff-9c6376c5ce46d9e39358812a3269ab6bR259

Reply:
You don't need anything `unsafe` to implement that at all- just add implement `spawn_obj` by inlining the body of `spawn_local_obj` and using `into` only if `upgrade` succeeds.

Reply:
You mean copy paste?